src/main/java/com/gregtechceu/gtceu/api/recipe/ingredient/nbtpredicate/EqualsNBTPredicate.java

```diff
@@ -0,0 +1,68 @@
+package com.gregtechceu.gtceu.api.recipe.ingredient.nbtpredicate;
+
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.NumericTag;
+import net.minecraft.nbt.Tag;
+
+import com.google.gson.JsonObject;
+
+import static com.gregtechceu.gtceu.api.recipe.ingredient.nbtpredicate.NBTPredicateUtils.getNestedTag;
+
+public class EqualsNBTPredicate extends NBTPredicate {
+
+    public static final String OP = "equals";
+
+    private String key;
+    private Tag value;
+    private boolean inverted;
+
+    public EqualsNBTPredicate(String key, Tag value) {
+        this(key, value, false);
+    }
+
+    public EqualsNBTPredicate(String key, Tag value, boolean inverted) {
+        this.key = key;
+        this.value = value;
+        this.inverted = inverted;
+    }
+
+    @Override
+    public boolean test(CompoundTag tag) {
+        Tag toCompare = getNestedTag(tag, key);
+        if (toCompare == null) {
+            return false;
+        } else {
+            // Mixed numeric types (e.g., int vs. double)
+            if (toCompare instanceof NumericTag toCompareNum &&
+                    value instanceof NumericTag valueNum) {
+                return inverted ^ toCompareNum.getAsDouble() == valueNum.getAsDouble();
+            }
+            return inverted ^ toCompare.equals(value);
+        }
+    }
+
+    public JsonObject toJson() {
+        JsonObject object = new JsonObject();
+        object.addProperty("op", OP);
+        object.addProperty("key", key);
+        object.add("value", NBTPredicateUtils.toJson(value));
+        object.addProperty("inverted", inverted);
+        return object;
+    }
+
+    public static NBTPredicate fromJson(JsonObject json) {
+        if (!json.get("op").getAsString().equals(OP)) {
+            throw new IllegalStateException("Trying to deserialize EqualsNBTPredicate but was something else: " + json);
+        }
+        if (!json.has("key") || !json.has("value") || !json.has("op")) {
+            throw new IllegalStateException("Could not deserialize EqualsNBTPredicate: " + json);
+        }
+        String key = json.get("key").getAsString();
+        Tag value = NBTPredicateUtils.fromJson(json.get("value"));
+        boolean inverted = false;
```

Comment:
Could be simplified by only writing an inverted property when it's true, then

```suggestion
        boolean inverted = json.has("inverted");
```